src/update-feeds/update-feeds.ts

```diff
@@ -124,8 +138,84 @@ export const runUpdateFeed = async (providerName: string, chain: Chain, chainId:
   });
 };
 
-// eslint-disable-next-line @typescript-eslint/require-await
-export const processBatch = async (batch: ReadDapiWithIndexResponse[]) => {
+export const mergeUrls = (receivedUrls: UrlSet[], freshExistingUrls: UrlSet[]) =>
+  Object.values(groupBy([...receivedUrls, ...freshExistingUrls], 'url')).flatMap(
+    (group) => sortBy(group, 'lastReceivedMs').pop()!
+  );
+
+export const updateDynamicState = (batch: ReadDapiWithIndexResponsesAndChainId) => {
```

Comment:
Why not just inline this?

Reply:
Yes, it was based on when I thought (and implemented) a scenario where we retained locally defined URLs.